### DISCUSSION TITLE
Reordering pages in a folder

### DISCUSSION THREAD

**Discussion by ShKhannanov** [requarks/wiki#5734] [Help / Questions] [answered]
Hey all,

is there any way to rearrange the order in which pages appear in a folder when using site tree navigation?

Thanks in advance!

**Comment by ghost** [comment]
Im really curious about that, too.

**Reply by ckingktg** [reply]
I'm having the same issue.... really hard to believe this is not an option.  does anyone have an answer on this for sorting pages on the nav in a custom order?

**Reply by BruceLiu507** [reply]
Can't agree more, it's baddly needed.

**Comment by NGPixel** [comment, marked as answer]
You can only sort pages when using a custom navigation. The browse mode will always list folders in alphanumerical order.

**Reply by griznog** [reply]
Foliant has an options for sorting pages that lets you prefix the page/directory name with `tag_` which it then strips from the name for displaying the site navigation. Is that something that could be added to wiki.js? It'd basically be an on/off option to strip everything up to and including the first occurrence of the chosen marker character for the displayed name.

**Reply by NGPixel** [reply]
This sounds like a really bad implementation idea so no. Use the custom navigation mode if you need to have a specific order.

**Reply by griznog** [reply]
It's one of the reasons I originally chose foliant. For a deeply nested site with a lot of pages it made it fairly trivial to preserve order with a simple prefix on the page filenames. But I can respect your view and it's not the end of the world to continue using a prefix and just have it show up in the name. But custom building the navigation by hand is definitely a non-starter.

**Reply by griznog** [reply]
Actually based on empirical testing, it seems that the site tree is having the names of folders modified for display already. If I use a `.` to delineate my sorting prefix from the rest of the directory name, the `.` is removed from the displayed name. But if I use a `-`, the `-` is preserved, for example I have directories for each section which are named like:

```
Section_1-Getting_Started
Section_2.Working_with_Slurm
```
and in the site navigation tree these show up as
```
Section_1-Getting_Started
Section_2Working_with_Slurm
```

Is that intentional or is this a bug? Again, not critical, for my purposes a `-` works just as well as a `.`, just thought it was worth noting the behavior. I haven't tested this with any other characters as my delimiter there.

**Comment by dreamerblue** [comment]
Same issue here.

Tree navigation cannot be customized in order, and custom navigation cannot display subpages (like a directory tree). Unfortunately, there is no perfect way to expand a tree list of documents in a controlled order that can be expanded infinitely.
